Sources/NIOSSH/Keys And Signatures/OpenSSHKey+Ed25519.swift

```diff
@@ -0,0 +1,106 @@
+import Foundation
+
+public enum OpenSSHKey {
+    public struct Ed25519Components {
+        public let seed: Data     // 32 bytes
+        public let publicKey: Data // 32 bytes
+    }
+
+    /// Decode an unencrypted OpenSSH Ed25519 private key (openssh-key-v1 format).
+    /// - Parameter pem: The PEM text including BEGIN/END OPENSSH PRIVATE KEY markers.
+    /// - Returns: Ed25519Components (seed + public key)
+    /// - Throws: If the key is not in the expected format or is encrypted.
+    public static func decodeEd25519Unencrypted(fromPEM pem: String) throws -> Ed25519Components {
```

Comment:
It would be useful to demonstrate that this does work. Can you wire up a unit test here?